tasks/e2e-monorepos.sh

```diff
@@ -110,25 +110,48 @@ pushd "$temp_app_path"
 cp -r "$root_path/packages/react-scripts/fixtures/monorepos/yarn-ws" .
 cd "yarn-ws"
 cp -r "$root_path/packages/react-scripts/fixtures/monorepos/packages" .
-yarn
 
 # Test cra-app1
-cd packages/cra-app1
+pushd packages/cra-app1
+cp -r "$root_path/packages/react-scripts/fixtures/monorepos/cra-app/"* .
+yarn
+verifyBuild
+yarn start --smoke-test
+verifyTest
+popd
+
+# ******************************************************************************
+# Test clean create-react-app inside workspace
+# ******************************************************************************
+pushd packages
+npx create-react-app newapp
+cd newapp
+yarn
+yarn build
+yarn start --smoke-test
+CI=true yarn test --watch=no
+popd
+
+# ******************************************************************************
+# Test create-react-app w/ shared comps inside workspace
+# ******************************************************************************
+pushd packages
+npx create-react-app --internal-testing-template="$root_path"/packages/react-scripts/fixtures/monorepos/cra-app cra-app2
+cd cra-app2
+yarn
 verifyBuild
 yarn start --smoke-test
 verifyTest
 
 # Test eject
+# TODO: veriy tests can be run from other apps after eject
+# -- will currently fail due to picking up ejected scripts/test.js
```

Comment:
Note, test.js in ejected app gets picked up (and fails because it's not a jest test) when running tests from other cra apps in the monorepo.  One way to solve this would be to detect other cra apps in the monorepo (pkgs with dependency on react-scripts?) and either not treat them as cra sources, or use approot/src as their source path instead of approot.  Another way could be by explicit cra source inclusion/exclusion.

Reply:
Can you explain in more detail? I don't follow. Whose test is this? Why is it not a jest test? What directory/package layout does this happen with?

Reply:
```
mono/
  cra-app1/
    scripts/
      test.js  <-- ejected test.js script, not a test!
    src/
      App.js
      App.test.js <-- a test
  cra-app2/
    src/
      App.js
```
When running "yarn test" in cra-app2, jest tries to run cra-app1/scripts/test.js as a test, same as cra-app1/src/App.test.js.  This is because we have given jest cra-app1 as a root and the jest testMatcher pattern '**/?(*.)(spec|test).{js,jsx,mjs}' matches test.js.

Just to be more clear -- this issue isn't related to the current PR, it just came up in testing this PR.  The issue comes up when you have multiple cra-apps in a monorepo, and one or more of them has been ejected.

Reply:
Looking at that structure, my earlier suggestion of detecting based on react-scripts as a dependency won't work.  Since that cra-app1 has been ejected, it doesn't have react-scripts as a dependency and wouldn't be detected as a cra app.  So, it falls into the category of being just some other thing in the monorepo that has its own build/test/etc.  I think we need some inclusion/exclusion config, instead of assuming everything in the monorepo is cra source.  (for the sake of posterity, this discussion probably belongs elsewhere -- should I create a new issue for the discussion?)

Reply:
Let's discuss in a new issue. 👍